### PR TITLE
Update  authController unit tests

### PR DESCRIPTION
**Pull request by msm-faraj**
This PR aims:
add custom jest matcher to compare errors
update unit test for authController

### CODE REVIEW

File: src/controllers/auth/authController.spec.js [hunk 1]
```diff
@@ -64,13 +65,13 @@ describe("AuthController", () => {
 
   describe("register", () => {
     describe("When everything is successful", () => {
-      beforeEach(async () => {
+      beforeEach(() => {
```
Comment (pouyapourbaba) [inline src/controllers/auth/authController.spec.js:68]: Keep the async/await here. Removing them might introduce some flaky behavior in the tests. We want to wait for the async method that we call to be executed then run our tests.

File: src/controllers/auth/authController.spec.js [hunk 2]
```diff
@@ -233,12 +225,12 @@ describe("AuthController", () => {
     });
 
     describe("when everything is successfull", () => {
-      beforeEach(async () => {
+      beforeEach(() => {
```
Comment (pouyapourbaba) [inline src/controllers/auth/authController.spec.js:228]: same here

File: src/controllers/auth/authController.spec.js [hunk 3]
```diff
@@ -400,7 +396,7 @@ describe("AuthController", () => {
           { authBcrypt: mockBcrypt }
         );
         mockAuthController.logOut(mockReq, mockRes);
-        await mockAuthController.logOut(mockReq, mockRes);
+        mockAuthController.logOut(mockReq, mockRes);
```
Comment (pouyapourbaba) [inline src/controllers/auth/authController.spec.js:399]: same here, but why di you call the logOut twice?

File: src/controllers/auth/authController.spec.js (hunk 1 above)
Comment (msm-faraj) [reply src/controllers/auth/authController.spec.js:68]: done

File: src/controllers/auth/authController.spec.js (hunk 3 above)
Comment (msm-faraj) [reply src/controllers/auth/authController.spec.js:399]: It was a mistake. I delete one of them. 
thanks

**Review by pouyapourbaba — APPROVED** [review]

**Pull request merged**